controllers/hostingdeploymentautoscalingjob/hostingdeploymentautoscalingjob_controller.go

```diff
@@ -0,0 +1,455 @@
+	}
+	ctx.Log.Info("Finalizer has been removed")
+
+	return err
+}
+
+// getResourceIDListfromInputSpec converts the list of resources into a string list to be used for various API calls
+func (r *Reconciler) getResourceIDListfromInputSpec(ctx *reconcileRequestContext) error {
+
+	resourceIDListfromSpec := ctx.HostingDeploymentAutoscalingJob.Spec.ResourceID
+	for _, resourceIDfromSpec := range resourceIDListfromSpec {
+		ResourceID := sdkutil.ConvertAutoscalingResourceToString(*resourceIDfromSpec)
+		ctx.ResourceIDList = append(ctx.ResourceIDList, *ResourceID)
+	}
+
+	// TODO: error handling
+	return nil
+}
+
+// Initialize fields on the context object which will be used later.
+func (r *Reconciler) initializeContext(ctx *reconcileRequestContext) error {
+	var err error
+
+	// Ensure we are using the job name specified in the spec
+	if ctx.HostingDeploymentAutoscalingJob.Spec.PolicyName != nil && len(*ctx.HostingDeploymentAutoscalingJob.Spec.PolicyName) > 0 {
+		ctx.PolicyName = *ctx.HostingDeploymentAutoscalingJob.Spec.PolicyName
+	} else {
+		ctx.PolicyName = DefaultAutoscalingPolicyName
+		ctx.HostingDeploymentAutoscalingJob.Spec.PolicyName = &ctx.PolicyName
+
+		// Review: Is this needed
+		if err := r.Update(ctx, ctx.HostingDeploymentAutoscalingJob); err != nil {
+			ctx.Log.Info("Error while updating HostingDeploymentAutoscalingJob policyName in spec")
+			return err
+		}
+	}
+
+	// Save the ResourceIDs into ctx as usable strings
+	if err = r.getResourceIDListfromInputSpec(ctx); err != nil {
+		ctx.Log.Error(err, "Error reading the ResourceIDs from Spec")
+		return err
+	}
+
+	// Initialize other values to defaults if not in Spec
+	if ctx.HostingDeploymentAutoscalingJob.Spec.ScalableDimension == nil {
+		namespace := sdkutil.HostingDeploymentAutoscalingServiceNamespace
+		ctx.HostingDeploymentAutoscalingJob.Spec.ServiceNamespace = &namespace
+	}
+
+	if ctx.HostingDeploymentAutoscalingJob.Spec.ScalableDimension == nil {
+		dimension := ScalableDimension
+		ctx.HostingDeploymentAutoscalingJob.Spec.ScalableDimension = &dimension
+	}
+
+	if ctx.HostingDeploymentAutoscalingJob.Spec.PolicyType == nil {
+		policyType := PolicyType
+		ctx.HostingDeploymentAutoscalingJob.Spec.PolicyType = &policyType
+	}
+
+	awsConfig, err := r.awsConfigLoader.LoadAwsConfigWithOverrides(*ctx.HostingDeploymentAutoscalingJob.Spec.Region, ctx.HostingDeploymentAutoscalingJob.Spec.SageMakerEndpoint)
+	if err != nil {
+		ctx.Log.Error(err, "Error loading AWS config")
+		return err
+	}
+
+	ctx.ApplicationAutoscalingClient = r.createApplicationAutoscalingClient(awsConfig)
+	ctx.Log.Info("Loaded AWS config")
+
+	return nil
+}
+
+// ctx.HostingDeploymentAutoscalingJob: desiredAutoscaling
+// ctx.ScalingPolicyDescription: actualAutoscaling
+func (r *Reconciler) determineActionForAutoscaling(ctx reconcileRequestContext) (controllers.ReconcileAction, error) {
+	var err error
+	// Review: Should you add a check based on status too, for controller restarts
```

Comment:
I will follow the hostingDeployment logic here nothing special.

Reply:
I followed the logic of model here.